### PR TITLE
Update client for gui version 21.2.0

### PR DESCRIPTION
**Pull request by niklasCarstensen**
Adds the valid_colors array and 
updates gamestate methods as well as rspec tests

### CODE REVIEW

**Review by SKoschnicke — APPROVED** [review]
I refactored some things. @niklasCarstensen please review this, too. (You only have to look at the last commit)

**Review by niklasCarstensen — COMMENTED** [review]
The refactor looks good overall 
but Im not sure if start_color needs to be initialized

File: lib/software_challenge_client/game_state.rb [hunk 1]
```diff
@@ -74,8 +82,7 @@ def field(x, y)
 
   # Erstellt einen neuen leeren Spielstand.
   def initialize
-    @start_color = Color::BLUE
```
[the remark this answers is not in the document]
Comment (SKoschnicke) [reply lib/software_challenge_client/game_state.rb]: start_color was set twice in your changes. I just removed one :)

Comment (niklasCarstensen) [reply lib/software_challenge_client/game_state.rb]: Ahh I overlooked that 😅

**Pull request merged**